tests/naive.py

```diff
@@ -772,28 +788,41 @@ def update(self, t):
 
 
 class stumpi_egress(object):
-    def __init__(self, T, m, excl_zone=None):
+    def __init__(self, T, m, excl_zone=None, k=1):
         self._T = np.asarray(T)
         self._T = self._T.copy()
         self._T_isfinite = np.isfinite(self._T)
         self._m = m
-        if excl_zone is None:
+        self._k = k
+
+        self._excl_zone = excl_zone
+        if self._excl_zone is None:
             self._excl_zone = int(np.ceil(self._m / config.STUMPY_EXCL_ZONE_DENOM))
 
         self._l = self._T.shape[0] - m + 1
-        mp = stump(T, m)
-        self.P_ = mp[:, 0]
-        self.I_ = mp[:, 1].astype(np.int64)
-        self.left_P_ = np.full(self.P_.shape, np.inf)
-        self.left_I_ = mp[:, 2].astype(np.int64)
-        for i, j in enumerate(self.left_I_):
-            if j >= 0:
-                D = core.mass(self._T[i : i + self._m], self._T[j : j + self._m])
+        mp = stump(T, m, exclusion_zone=self._excl_zone, k=self._k)
+        self.P_ = mp[:, :k].astype(np.float64)
+        self.I_ = mp[:, k : 2 * k].astype(np.int64)
+
+        self.left_I_ = mp[:, 2 * k].astype(np.int64)
+        self.left_P_ = np.full_like(self.left_I_, np.inf, dtype=np.float64)
+
+        for i, nn_i in enumerate(self.left_I_):
+            if nn_i >= 0:
+                D = core.mass(self._T[i : i + self._m], self._T[nn_i : nn_i + self._m])
```

Comment:
Wait a minute! This is in `naive.py`. We really should avoid calling `core.mass` here and, instead, we should use functions available to us in `naive.py`:

```
self.left_P_[i] = np.linalg.norm(
    z_norm(self._T[i : i + self._m]) - z_norm(self._T[nn_i : nn_i + self._m])
)
```

Reply:
So, if I remember correctly, the reason that kept it as `core.mass` is because `core.mass` behaves slightly differently. (I think I found this in one of my investigations but I am not sure though.)

If you follow the code in `core.mass`, you will end up with the function `core._calculate_squared_distance`. And, if you take a look at it, you can see it has some `if` blocks whose conditions are a function of the config variables. So, I think we need to create a **naive** version of function `mass` to behave the same. 

(For example, we can see we are doing something similar in naive `z_norm`. If you take a look at its function, it uses config variable to replace very small standard deviation with 1. We are doing the same config variable in `core.preprocess_diagonal` to modify std values.)

----

**ALSO, I JUST NOTE THAT:**
We need to fix `core.z_norm` and replace `std[std == 0]` with `std[std < config.STUMPY_STDDEV_THRESHOLD]`

Reply:
Maybe it's a problem but I feel like `naive.distance_profile` should be the same as `core.mass` except that `naive.distance_profile` is the "exact" version. Can you see what breaks if we replace `core.mass` with `naive.distance_profile` in `naive.py`? Then, we can document it and create a new issue to address it (and not solve it here).

> We need to fix core.z_norm and replace std[std == 0] with std[std < config.STUMPY_STDDEV_THRESHOLD]

Please do if you can!

Reply:
> Maybe it's a problem but I feel like `naive.distance_profile` should be the same as `core.mass` except that `naive.distance_profile` is the "exact" version. Can you see what breaks if we replace `core.mass` with `naive.distance_profile` in `naive.py`? Then, we can document it and create a new issue to address it (and not solve it here).

Sure.  will do so. 

> Please do if you can! Since it's small, we can add it to this PR or another (create a new issue). Your choice

I think I will fix it here in this PR.